lib/chef-acceptance/application.rb

```diff
@@ -4,26 +4,62 @@
 require "chef-acceptance/executable_helper"
 
 module ChefAcceptance
-  # Responsible for running one or more suites and displaying statistics about
-  # the runs to the user
+  class AcceptanceError < StandardError; end
+
+  # Responsible for running one or more suites and
+  # displaying statistics about the runs to the user.
   class Application
     attr_reader :force_destroy
+    attr_reader :output_formatter
 
     def initialize(options = {})
-      @force_destroy = options.fetch(:force_destroy, false)
+      @force_destroy = options.fetch("force_destroy", false)
+      @output_formatter = OutputFormatter.new
     end
 
     def run(suite, command)
+      start_time = Time.now
+      error = false
+      run_summary = true
+
+      begin
+        run_suite(suite, command)
+      rescue AcceptanceError => e
+        error = true
+        run_summary = false
+        puts "#{e.to_s}\n#{e.backtrace.join("\n")}"
+      rescue RuntimeError
+        # We catch the errors here and do not raise again in
+        # order to make a clean exit. Since the errors are already
+        # in stdout, we do not print them again.
+        error = true
+      ensure
+        if run_summary
+          total_duration = Time.now - start_time
+          # Special footer row that gives overall status
+          output_formatter.add_row(suite: "", command: "Total", duration: total_duration, error: error)
+
+          puts ""
+          puts "chef-acceptance run #{error ? "failed" : "succeeded"}."
+          puts output_formatter.generate_output
+        end
+      end
+
+      # Make sure that exit code reflects the error if we have any
+      exit(1) if error
```

Comment:
What do you think about raising an AcceptanceError instead and letting the Cli class handle the exception and call exit(1)?